### PR TITLE
Fixed Distinct Encoding Bug

### PR DESCRIPTION
**Pull request by bchong95**
# Fixed Distinct Encoding Bug

## Description

DISTINCT queries had a bug when encoding unicode characters. Encoding.Utf8.GetBytes was being given the length in utf8 characters instead of the char count of the string.

## Type of change

Please delete options that are not relevant.

- [ x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Query/Core/DistinctMap.UnorderedDistinctMap.cs [hunk 1]
```diff
@@ -377,7 +377,7 @@ private bool AddStringValue(string value)
                 {
                     // Zero out the array since you want all trailing bytes to be 0 for the conversions that happen next.
                     Array.Clear(this.utf8Buffer, 0, this.utf8Buffer.Length);
-                    Encoding.UTF8.GetBytes(value, 0, utf8Length, this.utf8Buffer, 0);
+                    Encoding.UTF8.GetBytes(value, 0, value.Length, this.utf8Buffer, 0);
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Query/Core/DistinctMap.UnorderedDistinctMap.cs]: Test?

File: changelog.md [hunk 2]
```diff
@@ -30,6 +30,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - [#1036](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/1036) Fixed query responses to return null Content if it is a failure
 - [#1045](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/1045) Added stack trace and innner exception to CosmosException
 - [#1050](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/1050) Add mocking constructors to TransactionalBatchOperationResult
+- [#1060](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/1060) Fixed encoding bug in DISTINCT queries.
```
Comment (j82w) [inline changelog.md]: Thank you

Comment (kirankumarkolli) [reply changelog.md]: @j82w this also needs to be pulled into 3.5.0

Comment (kirankumarkolli) [reply changelog.md]: @bchong95 can we please rephrase to convey more about the bug. Ex: "unicode encoding bug", "non ASCII encoding..."

File: Microsoft.Azure.Cosmos/src/Query/Core/DistinctMap.UnorderedDistinctMap.cs (hunk 1 above)
Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/Query/Core/DistinctMap.UnorderedDistinctMap.cs]: Added a quick test.

File: changelog.md (hunk 2 above)
Comment (bchong95) [reply changelog.md]: Done.

File: changelog.md [hunk 3]
```diff
@@ -30,6 +30,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - [#1036](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/1036) Fixed query responses to return null Content if it is a failure
 - [#1045](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/1045) Added stack trace and innner exception to CosmosException
 - [#1050](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/1050) Add mocking constructors to TransactionalBatchOperationResult
+- [#1060](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/1060) Fixed unicode encoding bug in DISTINCT queries.
```
Comment (j82w) [inline changelog.md]: 3.5.0 is shipped please move this up to unreleased

**Pull request merged**